### PR TITLE
CLDR-13916 Update help link for unitLength...gender; make clickable

### PR DESCRIPTION
**Pull request by btangmu**
-Add CLDRURLS.GRAMMATICAL_INFLECTION for .../unitLength.../gender paths

-Make helpHtml links clickable in Info Panel

-Remove dead code from PathDescription.java

CLDR-13916

- [x] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Review by macchiati — COMMENTED** [review]
Looks ok, but Steven should look at the mjs

File: tools/cldr-apps/js/src/esm/cldrDeferHelp.mjs [hunk 1]
```diff
@@ -8,6 +8,9 @@ const abstractLang = "en";
 const USELESS_IN_INFO_PANEL = "see info panel";
 
 function addDeferredHelpTo(fragment, helpHtml, resource, translationHint) {
+  if (helpHtml) {
+    helpHtml = linkifyHttpsUrls(helpHtml);
```
Comment (srl295) [inline tools/cldr-apps/js/src/esm/cldrDeferHelp.mjs:12]: fine for now, pending a better solution in https://unicode-org.atlassian.net/browse/CLDR-15708

File: tools/cldr-apps/js/src/esm/cldrDeferHelp.mjs [hunk 2]
```diff
@@ -45,6 +48,16 @@ function addDeferredHelpTo(fragment, helpHtml, resource, translationHint) {
...
+function linkifyHttpsUrls(text) {
+  const regex = /\b(https:\/\/\S+)/g;
+  const s = text.replace(regex, "<a target='_blank' href='$1'>$1</a>");
+  // In principle, a URL can end in a period;
+  // in practice, a final period in PathDescription.java is not part of the URL
+  const periodQuoteEndBracketRegex = /\.'>/g;
+  const quoteEndBracket = "'>";
+  return s.replace(periodQuoteEndBracketRegex, quoteEndBracket);
+}
+
```
Comment (srl295) [inline tools/cldr-apps/js/src/esm/cldrDeferHelp.mjs:60]: lots of problems here.  OK for now, pending https://unicode-org.atlassian.net/browse/CLDR-15708

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/PathDescription.java [hunk 3]
```diff
@@ -1025,32 +1023,6 @@ public String getDescription(String path, String value, Object context) {
         return description;
     }
 
-    /**
-     * Creates an escaped HTML string of placeholder information.
-     *
-     * @param path the xpath to specify placeholder information for
-     * @return a HTML string, or an empty string if there was no placeholder information
-     */
-    public String getPlaceholderDescription(String path) {
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/util/PathDescription.java:1034]: huh, i wonder when this got dropped?  Probably accidental.

This was added in CLDR-4544, and then unknowingly replicated in https://github.com/unicode-org/cldr/pull/2012

**Review by srl295 — APPROVED** [review]
It's OK for now. I'm not at all a fan of auto-linkification.

File: tools/cldr-apps/js/src/esm/cldrDeferHelp.mjs (hunk 1 above)
Comment (btangmu) [reply tools/cldr-apps/js/src/esm/cldrDeferHelp.mjs:12]: I agree about auto-linkification not being the best way, still I think it's better than expecting ST users to copy and paste links. Perfect is the enemy of good.

Comment (srl295) [reply tools/cldr-apps/js/src/esm/cldrDeferHelp.mjs:12]: Yes, and I'm glad to approve this and improve things for users.

**Pull request merged**